src/bencode/core.clj

```diff
@@ -314,14 +314,15 @@
   'namespace/name'."
   (fn [_output thing]
     (cond
-      (instance? (RT/classForName "[B") thing) :bytes
+      (nil? thing) :list
+      (-> thing class .getComponentType (= Byte/TYPE)) :bytes
       (instance? InputStream thing) :input-stream
       (integer? thing) :integer
       (string? thing)  :string
       (symbol? thing)  :named
       (keyword? thing) :named
       (map? thing)     :map
-      (or (nil? thing) (coll? thing) (.isArray (class thing))) :list
+      (or (coll? thing) (.isArray (class thing))) :list
```

Comment:
This change is unrelated, so probably it should be in a different commit, so people won't think it's something to do with GraalVM.

Reply:
That change is not unrelated though. As explained, the nil check was moved up, since we needed that in the check for bytes.